unit_test/test_pub_9.py

```diff
@@ -0,0 +1,19 @@
+import sys
+sys.path.insert(1, '../')
+
+import unittest
+import nttt
+
+
+class TestPub9(unittest.TestCase):
+    def test_pub_9(self):
+    
+        c_initial = 'asd _ fgh _ asd ` ghj ` asd ** hjk ** asd * uio * asd'
+        c_target = 'asd _fgh_ asd `ghj` asd **hjk** asd *uio* asd'
+        
+        self.assertEqual(nttt.utilities.trim_pub_9(c_initial), c_target)
+    
+
```

Comment:
Other cases that could be tested:
c_initial = 'asd _ fgh_ asd ` ghj` asd ** hjk** asd * uio* asd' (only spaces after the open tag)
c_initial = 'asd _fgh _ asd `ghj ` asd **hjk ** asd *uio * asd' (only spaces before the close tag)
c_initial = 'asd _  fgh  _ asd `  ghj  ` asd **  hjk  ** asd *  uio  * asd' (multiple spaces)

Reply:
OK with left and right spaces.

I have problems implementing multiple spaces. There are some complications.

Reply:
I added some more tests that reveal that there are more problems.
For example
- asd _   fgh _ asd _ ghj   _ asd
+ asd _fgh_ asd _ghj_ asd

It fails by converting it all to 
- asd _fgh_asd_ghj_ asd

The problem here is that underscores are not like eg xml markup where you know which is the beginning and which the ending tag. All underscores are identical for the regular expression. So how can we fix fgh and ghj without affecting asd? Any ideas?

Reply:
Fixed. I added some more tests to be sure.